### ISSUE TITLE
AppBarThemeCompactHeight requires retemplating to look appropriate 

### ISSUE THREAD

**Issue by StephenLPeters** [microsoft/microsoft-ui-xaml#1424]
> ![image](https://user-images.githubusercontent.com/4424330/66523183-1a66ed80-eaa4-11e9-93ce-119ca5835ab4.png)
> @chingucoding I think we also need to center the TextLabel.

Yes and no. As @licanhua mentioned [in this comment](https://github.com/microsoft/microsoft-ui-xaml/issues/275#issuecomment-463767698), when one changes the `AppBarThemeCompactHeight` , one also has to changes other values to get everything to align correctly. So this would be sort of by design I guess.

Since the text is positioned using margins, I think we need to re-template the positioning of the label. And this, I think will, be quite hard, as a lot of fine tuning was needed to get the app bar button to look the way it does (according to @jevansaks and @licanhua).

_Originally posted by @chingucoding in https://github.com/microsoft/microsoft-ui-xaml/pull/1411#issuecomment-540220055_

The app bar button's text being off center and requiring a retemplate to fix means that this light weight styling api is essentially broken.

**Comment by StephenLPeters** [comment]
#1411  is related

**Comment by jevansaks** [comment]
Is anyone blocked on using normal density or compact density? Or did we just find that this customization point doesn't work well?

**Comment by chingucoding** [comment]
> Is anyone blocked on using normal density or compact density? Or did we just find that this customization point doesn't work well?

Both normal and compact should work now (I think). Customization of the AppBar(Toggle)Button however is very difficult and changing simple properties such as height essentially requires settings multiple values instead of simple setting height. 

So while the component works, customization is definitely not as easy as one would expect.

**Comment by jevansaks** [comment]
Duplicate of #275.
